File: src/main/java/io/github/dqualizer/dqlang/archive/k6adapter/dqlang/loadtest/LoadTest.java
package io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.domain_architecture_mapping.Endpoint;
import io.github.dqualizer.dqlang.types.rqa.definition.Artifact;
import io.github.dqualizer.dqlang.types.rqa.definition.ResponseMeasure;
import lombok.*;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class LoadTest {

    private Artifact artifact;
    @NonNull private String description;
    private Stimulus stimulus;
    @JsonProperty("response_measure")
    private ResponseMeasure responseMeasure;
    private Endpoint endpoint;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/k6adapter/dqlang/loadtest/LoadTest.kt
package io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
class LoadTest {
    val artifact: io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest.Artifact? = null
    val description: String? = null
    val stimulus: io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest.Stimulus? = null

    @JsonProperty("response_measure")
    val responseMeasure: io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest.ResponseMeasure? = null
    val endpoint: io.github.dqualizer.dqlang.archive.k6adapter.dqlang.loadtest.Endpoint? = null
}
